src/app/jscaip/tests/RulesUtils.spec.ts

```diff
@@ -2,61 +2,102 @@ import { LegalityStatus } from '../LegalityStatus';
 import { MGPNode } from '../MGPNode';
 import { Minimax } from '../Minimax';
 import { Move } from '../Move';
-import { NodeUnheritance } from '../NodeUnheritance';
 import { Player } from '../Player';
 import { GameStatus, Rules } from '../Rules';
 import { AbstractGameState } from '../GameState';
+import { comparableEquals, ComparableObject } from 'src/app/utils/Comparable';
+export class RulesUtils {
 
-export function expectToBeVictoryFor<M extends Move,
-                                     S extends AbstractGameState,
-                                     L extends LegalityStatus,
-                                     U extends NodeUnheritance>(
-    rules: Rules<M, S, L>,
-    node: MGPNode<Rules<M, S, L>, M, S, L, U>,
-    player: Player,
-    minimaxes: Minimax<M, S>[])
-: void
-{
-    expect(rules.getGameStatus(node)).toEqual(GameStatus.getVictory(player));
-    for (const minimax of minimaxes) {
-        expect(minimax.getBoardValue(node).value)
-            .withContext(minimax.name + ' should consider part a victory for player ' + player.value)
-            .toEqual(player.getVictoryValue());
+    public static expectMoveSuccess(rules: Rules<Move, AbstractGameState>,
+                                    state: AbstractGameState,
+                                    move: Move,
+                                    expectedState: AbstractGameState)
+    : void
+    {
+        const legality: LegalityStatus = rules.isLegal(move, state);
+        expect(legality.legal).toBeTruthy();
+        if (legality.legal.isSuccess()) {
+            const resultingState: AbstractGameState = rules.applyLegalMove(move, state, legality);
+            if (resultingState['equals'] !== null) { // TODOTODO: will be isComparableObject when your branch is merged
```

Comment:
cependant, l'avantage de pas appeler `comparableEquals` et de faire du `.toEquals` c'est d'avoir ceci par exemple, plutôt que "expected false to be true":
```
	states should be equal:
	   Expected $.board[0][4].owner.value = 0 to equal 2.
	   Expected $.board[3][0].owner.value = 2 to equal 0.
	Error: states should be equal:
	   Expected $.board[0][4].owner.value = 0 to equal 2.
	   Expected $.board[3][0].owner.value = 2 to equal 0.```

Reply:
or simply `isComparable`

Reply:
Ouais c'est vrai c'est déjà des bons outils, mais si l'un est pas comparable, pouvoir utiliser .equals permet d'avoir plusieurs options, mais oui pour l'instant ça marche très bien avec juste expect(a).toEquals(b) (ça couvre juste 4 branches en moins)

Reply:
je résouds car y'a pas d'action à prendre ici